server/game/core/card/propertyMixins/UnitProperties.ts

```diff
@@ -299,15 +301,26 @@ export function WithUnitProperties<TBaseClass extends InPlayCardConstructor>(Bas
         }
 
         // ***************************************** STAT HELPERS *****************************************
-        public override addDamage(amount: number): void {
-            super.addDamage(amount);
+        public override addDamage(amount: number, source: INonFrameworkDamageOrDefeatSource): void {
+            super.addDamage(amount, source);
 
-            this.checkDefeated();
+            this.checkDefeated(source);
         }
 
-        public checkDefeated() {
+        // TODO: FFG has yet to release detailed rules about how effects like Snoke are used to determine which player defeated a unit.
```

Comment:
Oh yeah I need to update this comment, you're right that Snoke is a bad example since there is an official ruling here. Great callout.

What I was really thinking of are more complex situations where the "chain" of effects is longer - for example, what if there is a Dodonna effect keeping a Rebel unit alive with the +1 hp and then the Dodonna is killed? Whatever we implement needs to be able to handle all of those situations, so instead of trying to make something I figure we'll just wait for the new ruleset.